src/icons/Status.js

```diff
@@ -1,46 +1,143 @@
 import React from 'react'
+import PropTypes from 'prop-types'
+import {
+    iconStyle,
+    iconStyleDefault,
+    iconStyleValid,
+    iconStyleWarning,
+    iconStyleError,
+} from './styles'
 
-export function Valid({ className }) {
+const stylesPropType = PropTypes.shape({
+    className: PropTypes.string.isRequired,
+    styles: PropTypes.object.isRequired,
+})
+
+/**
+ * Abstract icon component
+ *
+ * @param {Object} props
+ * @param {Object} props.styles
+ * @param {any} props.children
+ * @returns {Object}
+ */
+function Icon({ styles, children, className }) {
     return (
         <svg
             xmlns="http://www.w3.org/2000/svg"
             width="48"
             height="48"
             viewBox="0 0 48 48"
-            className={className}
+            className={[iconStyle.className, styles.className, className].join(
+                ' '
+            )}
         >
-            <path d="M0 0h48v48H0z" fill="none" />
-            <path d="M24 4C12.95 4 4 12.95 4 24c0 11.04 8.95 20 20 20 11.04 0 20-8.96 20-20 0-11.05-8.96-20-20-20zm-4 30L10 24l2.83-2.83L20 28.34l15.17-15.17L38 16 20 34z" />
+            {children}
+            <style>{iconStyle.styles}</style>
```

Comment:
The tag should be `<style jsx>`. Then you should be able to switch to:

```
const iconStyle = css`
    svg {...}
`

<style jsx>{iconStyle}</style>
```

Reply:
see comment above, depends on what we agree on there